src/model.ts

```diff
@@ -1010,6 +1011,26 @@ class Model<T> extends EventEmitter {
     return JSON.stringify(this.toJSON());
   }
 
+  async toJSONStream(writeStream: Writable): Promise<void> {
+    let p: Promise<unknown> | undefined;
+    const { data, schema } = this;
+    const keys = this.dataKeys;
+    const { length } = keys;
+
+    p = asyncWriteToStream(writeStream, '[');
+    if (p) await p;
+    for (let i = 0; i < length; i++) {
+      const raw = data[keys[i]];
+      if (raw) {
+        const prefix = i === 0 ? '' : ',';
+        p = asyncWriteToStream(writeStream, `${prefix}${JSON.stringify(schema._exportDatabase(cloneDeep(raw) as object))}`);
```

Comment:
This new function writes each cloned object directly to stream, making it more friendly for escape analysis